gcloud-java-core/src/main/java/com/google/gcloud/BaseServiceException.java

```diff
@@ -62,16 +68,25 @@ public Integer code() {
     }
 
     /**
+     * Returns true if the error indicates that the API call was certainly not accepted by the
+     * server.
+     */
+    public boolean rejected() {
+      return rejected;
+    }
+
+    /**
      * Returns the reason that caused the exception.
      */
     public String reason() {
       return reason;
     }
 
-    boolean isRetryable(Set<Error> retryableErrors) {
+    boolean isRetryable(boolean idempotent, Set<Error> retryableErrors) {
       for (Error retryableError : retryableErrors) {
-        if ((retryableError.code() == null || retryableError.code().equals(this.code()))
-            && (retryableError.reason() == null || retryableError.reason().equals(this.reason()))) {
+        boolean match = (retryableError.code() == null || retryableError.code().equals(this.code()))
+            && (retryableError.reason() == null || retryableError.reason().equals(this.reason()));
+        if (match && idempotent || match && retryableError.rejected()) {
```

Comment:
I would leave the condition as it was and just instead of `return true` have `return idempotent || retryableError.rejected()`

There is no reason to continue the loop once we have a match.